XenAdmin/Diagnostics/Checks/PoolHasDeprecatedSrsCheck.cs

```diff
@@ -0,0 +1,58 @@
+﻿/* Copyright (c) Citrix Systems, Inc. 
+ * All rights reserved. 
+ * 
+ * Redistribution and use in source and binary forms, 
+ * with or without modification, are permitted provided 
+ * that the following conditions are met: 
+ * 
+ * *   Redistributions of source code must retain the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer. 
+ * *   Redistributions in binary form must reproduce the above 
+ *     copyright notice, this list of conditions and the 
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System.Linq;
+using XenAdmin.Core;
+using XenAPI;
+using XenAdmin.Diagnostics.Problems;
+using XenAdmin.Diagnostics.Problems.PoolProblem;
+
+namespace XenAdmin.Diagnostics.Checks
+{
+    internal class PoolHasDeprecatedSrsCheck : HostPostLivenessCheck
+    {
+        public override string Description => string.Format(Messages.DEPRECATED_SRS_CHECK, Helpers.GetPoolOfOne(Host.Connection));
+
+        public PoolHasDeprecatedSrsCheck(Host host)
+            : base(host)
+        {
+        }
+
+        public override bool CanRun() => !Helpers.Post82X(Host.Connection);
+
+        protected override Problem RunHostCheck()
+        {
+            var hasFCoESr = Host.Connection.Cache.SRs.Any(sr => sr.GetSRType(true) == SR.SRTypes.lvmofcoe);

```

Comment:
We need to check the target version and decide accordingly what to show. The check is done with the upgrade plugin for automated mode. For now we'll show a warning only for >= Post82X. In manual mode or when the aforementioned plugin has failed and we don't know the target version, we need to show a conditional warning. (In future we'll add a blocking error if IsYOrGreater in automated mode and keep the warning for manual; this will also depend on the upgrade compatibility matrix, but we don't need to worry about it now).

Reply:
![image](https://user-images.githubusercontent.com/32554698/218706340-07757505-7f4f-4df7-9ecb-da0294e41556.png)